### PR TITLE
Simplify example for lifecycle callback in release notes

### PR DESCRIPTION
**Pull request by abstratt**
Fixes: #29044

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

File: platforms/documentation/docs/src/docs/release/notes.md [hunk 1]
```diff
@@ -190,20 +190,21 @@ This will allow Gradle to perform additional performance optimizations and will
...
 
-def start = 0
 gradle.lifecycle.beforeProject {
-    start = System.currentTimeMillis()
-}
-gradle.lifecycle.afterProject { project ->
-    def end = System.currentTimeMillis()
-    def elapsed = end - start
-    println "Project $project.name configured in ${elapsed}ms"
+    if (this.path != this.rootProject.path) {
```
Comment (bamboo) [inline platforms/documentation/docs/src/docs/release/notes.md:202]: ❓ Do we need to complicate the example with a conditional here? I'd just keep the body of the `if`, but if you feel strongly about showing it's possible to read project properties I'd ditch the `this.` and go for the idiomatic `if (parent != null)`

Comment (abstratt) [reply platforms/documentation/docs/src/docs/release/notes.md:202]: Not really, will get rid of it, and change the description

**Review by bamboo — APPROVED** [review]
Nice! 👍

File: platforms/documentation/docs/src/docs/release/notes.md [hunk 2]
```diff
@@ -215,10 +214,10 @@ The view exposes only those properties that are safe to access across project bo
...
 
-```groovy
-gradle.lifecycle.beforeProject { project ->
-   def rootDir = project.isolated.rootProject.projectDirectory
-   println "The root project directory is $rootDir"
+```kotlin
+gradle.lifecycle.beforeProject {
+    val rootDir = project.isolated.rootProject.projectDirectory
+    println("The root project directory is $rootDir")
 }
```
Comment (alllex) [inline platforms/documentation/docs/src/docs/release/notes.md:221]: ❓ Would dropping the `beforeProject` call make more sense?
The sample in full is not useful: it prints the same line as many times as there are subprojects.

One of the real benefits of the isolated view is that you can use it from a subproject's build script.
```
subproject/build.gradle.kts

checkstyle {
    val rootDir = project.isolated.rootProject.projectDirectory
    configDirectory = rootDir.dir("checkstyle")
}
```

**Pull request merged**